scripts/west_commands/runners/up_squared.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2017 Linaro Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing on up squared board via pxe boot.'''
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+
+class UpSquaredBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for Up Squared.'''
+    # This is an special flash command for up squared board.
+    # It will invoke the pxe_boot_sh script set by '--pxe-boot'
+    # option to complete the test case image downloading for
+    # up_squared board via pxe boot.
+
+    def __init__(self, cfg, pxe_boot_sh):
+        super(UpSquaredBinaryRunner, self).__init__(cfg)
+        self.pxe_boot_sh = pxe_boot_sh
+
+    @classmethod
+    def name(cls):
+        return 'up_squared'
+
+    @classmethod
+    def capabilities(cls):
+        # only have to support the flash command.
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        # using the pxe-boot.sh script to achieve the goal copying
+        # test case iamge and grub image to board
+        parser.add_argument('--pxe-boot', required=True,
+                            help='pxe boot control command')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return UpSquaredBinaryRunner(cfg, args.pxe_boot)
+
+    def do_run(self, command, **kwargs):
+        if self.pxe_boot_sh is None:
+            raise ValueError('Cannot flash; --pxe-boot not given.')
+        # test case image "zephyr.strip"
+        strip_file = self.cfg.elf_file[:-4] + ".strip"
+        # run pxe_boot_sh script
+        cmd = [self.pxe_boot_sh, strip_file]
```

Comment:
You need to define the interface of that script somewhere, so whoever is going to implement it knows what to expect from west and how to write this script.
I also do not see the script, an example or a template for the script that could work with this runner would be very useful.

Reply:
I will add a section in the boards/x86/up_squared/doc/index.rst to introduce this, there will be a script template.